### PR TITLE
split system + material component

### PR DESCRIPTION
**Pull request by ngokevin**
**Description:**

Precursor to doing changes in https://github.com/aframevr/aframe/pull/1364

This is mostly moving code around

**Changes proposed:**
- Split material and texture system
- Rename textureCache -> cache

### CODE REVIEW

File: src/systems/template.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+var registerSystem = require('../core/system').registerSystem;
+
+/**
+ * Template system.
+ *
+ * Handles lazy-loading template engines, template compilation, caching of compiled templates.
+ */
+module.exports.System = registerSystem('template', {
```
Comment (dmarcos) [inline src/systems/template.js]: I think we don't need this file

File: src/utils/material.js [hunk 2]
```diff
@@ -10,16 +10,16 @@ module.exports.updateMap = function (shader, data) {
   var el = shader.el;
   var src = data.src;
   var material = shader.material;
-  var materialSystem = el.sceneEl.systems.material;
+  var textureSystem = el.sceneEl.systems.texture;
```
Comment (dmarcos) [inline src/utils/material.js]: Why making texture a system and not keeping it as utils library?

File: src/systems/template.js (hunk 1 above)
Comment (ngokevin) [inline src/systems/template.js]: oops

File: src/utils/material.js (hunk 2 above)
Comment (ngokevin) [inline src/utils/material.js]: It's a bit more consistent to use systems when we want components to access some sort of global scope. Also would be nice to inspect the cache ```sceneEl.systems.texture.cache```.

Comment (dmarcos) [inline src/utils/material.js]: I'm really not sure about it. Why not making all the utils systems then? Keeping the 1:1 relationship between component and system is important . 1 component type should be paired with one system or you will be coupling the state of components of different types via those shared systems (scopes). Systems should not be something you can require or there's no difference between a system and a library.

Comment (ngokevin) [inline src/utils/material.js]: The only reason I can think of is that a system is specific to a single A-Frame scene, and a utility module is global among all A-Frame scenes.

I'll keep everything in the material system then. I think that doesn't hurt.

**Pull request closed without merging**